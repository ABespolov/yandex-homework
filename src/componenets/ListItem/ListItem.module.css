.container {
  position: relative;
  display: block;
  margin-bottom: 12px;
  padding-left: 35px;
  font-size: 22px;
  cursor: pointer;
  user-select: none;
}

.container input {
  position: absolute;
  width: 0;
  height: 0;
  cursor: pointer;
  opacity: 0;
}

.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  width: 25px;
  height: 25px;
  background-color: #d9d9d9;
  border-radius: 3px;
}

.checked input ~ .checkmark {
  background-color: #fc0;
}

.container:hover input ~ .checkmark {
  background-color: #fff;
}

.checkmark::after {
  content: "";
  position: absolute;
  display: none;
}

.checked .checkmark::after {
  top: 5px;
  left: 9px;
  width: 5px;
  height: 10px;
  border: solid #000;
  border-width: 0 3px 3px 0;
  transform: rotate(45deg);
}

.checked input ~ .checkmark::after {
  display: block;
}

.checked {
  text-decoration: line-through;
}
