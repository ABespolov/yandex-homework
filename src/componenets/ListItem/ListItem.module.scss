.container {
  position: relative;
  display: block;
  margin-bottom: 12px;
  padding-left: 30px;
  font-size: 18px;
  line-height: 18px;
  cursor: pointer;
  user-select: none;
}

.container input {
  position: absolute;
  width: 0;
  height: 0;
  cursor: pointer;
  opacity: 0;
}

.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  width: 18px;
  height: 18px;
  background-color: #d9d9d9;
  border-radius: 3px;
}

.checked input ~ .checkmark {
  background-color: #fc0;
}

.container:hover input ~ .checkmark {
 // background-color: #fff;
}

.checkmark::after {
  content: "";
  position: absolute;
  display: none;
}

.checked .checkmark::after {
  top: 3px;
  left: 6px;
  width: 5px;
  height: 8px;
  border: solid #000;
  border-width: 0 2px 2px 0;
  transform: rotate(45deg);
}

.checked input ~ .checkmark::after {
  display: block;
}

.checked {
  margin-bottom: 0;
  text-decoration: line-through;
}
