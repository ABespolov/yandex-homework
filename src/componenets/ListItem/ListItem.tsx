import React from "react";
import styles from "./ListItem.module.css";

interface Items {
  text: string;
  checked: boolean;
}

interface ListItem {
  item: Items;
  isChecked: boolean;
}

export const ListItem: React.FC<ListItem> = ({ item, isChecked }) => {
  return (
    <label htmlFor={item.text} className={`${styles.container} ${isChecked ? styles.checked : ""}`}>
      {item.text}
      <input id={item.text} type="checkbox"/>
      <span className={styles.checkmark} />
    </label>
  );
};
