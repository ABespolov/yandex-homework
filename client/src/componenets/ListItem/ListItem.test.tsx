import {configure, shallow} from "enzyme";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import {ListItem} from "./ListItem";

configure({ adapter: new Adapter() });

describe("ListItem:", () => {
    it("ListItem render: ", () => {
        const items = {text: 'text', checked: false};
        const component = shallow(<ListItem item={items} isChecked={false} />);
        expect(component).toMatchSnapshot();
    });
});
