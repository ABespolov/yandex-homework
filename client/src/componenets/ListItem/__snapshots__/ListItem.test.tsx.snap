// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem: ListItem render:  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    isChecked={false}
    item={
      Object {
        "checked": false,
        "text": "text",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "text",
        <input
          id="text"
          type="checkbox"
        />,
        <span
          className="checkmark"
        />,
      ],
      "className": "container ",
      "htmlFor": "text",
    },
    "ref": null,
    "rendered": Array [
      "text",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "text",
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "checkmark",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "text",
          <input
            id="text"
            type="checkbox"
          />,
          <span
            className="checkmark"
          />,
        ],
        "className": "container ",
        "htmlFor": "text",
      },
      "ref": null,
      "rendered": Array [
        "text",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "text",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "checkmark",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
